plugins/woocommerce-blocks/assets/js/blocks/checkout/inner-blocks/checkout-billing-address-block/edit.tsx

```diff
@@ -61,6 +61,11 @@ export const Edit = ( {
 		forcedBillingAddress
 	);
 
+	// This is needed to force the block to re-render when the requireApartmentField changes.
+	const blockKey = `billing-address-${
+		requireApartmentField ? 'visible' : 'hidden'
+	}-address-2`;
+
```

Comment:
Changing a component key force it to remount, which forces the state initial value to recompute.

Reply:
Thanks for applying this change, @senadir.